File: leetcode_problems/python/lc300_longest_increasing_subsequence.py
class Solution:
    def lengthOfLIS(self, nums: list[int]) -> int:
        if len(nums) == 1:
            return 1

        n = len(nums)
        dp = [1] * n
        for i in range(1, n):
            for j in range(0, i):
                if nums[i] > nums[j]:
                    dp[i] = max(dp[i], dp[j] + 1)
        return max(dp)


if __name__ == "__main__":
    nums = [10, 9, 2, 5, 3, 7, 101, 18]
    print(Solution().lengthOfLIS(nums))

File: leetcode_problems/typescript/src/lc300_longest_increasing_subsequence.ts
function lengthOfLIS(nums: number[]): number {
    let n = nums.length;
    let dp = new Array(n).fill(1);
    for (let i=1;i<n;i++) {
        for (let j=0;j<i;j++) {
            if (nums[i] > nums[j]) {
                dp[i] = Math.max(dp[i], dp[j] + 1);
            }
        }
    }
    let max = 0;
    for (let num of dp) {
        max = Math.max(num, max);
    }
    return max;
};